specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,383 @@
+
+    /// Context Menu Items displayed by the Edge browser, holds the properties of a context menu item
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown {
+        /// Get the Name displayed for the Context Menu Item
+        [propget] HRESULT Name([out, retval] LPWSTR* name);
+
+        /// Get the ID for the Context Menu Item
+        [propget] HRESULT ID([out, retval] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID* id);
+
+        /// Get the Shortcut for the Context Menu Item (CTRL+X for Cut)
+        [propget] HRESULT Shortcut([out, retval] LPWSTR* short_cut);
+
+    }
+
+    /// Collection of ContextMenuItem objects 
+    [uuid(f562a2f5-c415-45cf-b909-d4b7c1e276d3), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItemCollection : IUnknown {
+        /// Gets the iterator over the collection of context menu items from the start
+        HRESULT GetIterator([out, retval] ICoreWebView2ContextMenuItemCollectionIterator ** iterator);
+
+        /// Returns whether the specific Context Menu ID is contained in the collection
+        HRESULT Contains([in] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID menuItem, [out, retval] BOOL * iterator);
+
+        /// Returns the iterator starting at the menu item specified
+        HRESULT Get([in] COREWEBVIEW2_CONTEXT_MENU_ITEM_ID menuItem, [out, retval]  ICoreWebView2ContextMenuItemCollectionIterator ** iterator);
+
+    }
+
+    /// Iterator for a collection of ContextMenuItem objects 
+    [uuid(b9cc0ce5-4d10-4b9c-af06-802b199abc42), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItemCollectionIterator : IUnknown {
+        /// TRUE when the iterator has not run out of ContextMenuItem objects
+        /// Will be FALSE if collection is empty or if iterator has gone past collection
+        [propget] HRESULT HasCurrent([out, retval] BOOL * hasCurrent);
+
+        /// Gets the current ICoreWebView2ContextMenuItem of the iterator
+        HRESULT GetCurrent([out, retval] ICoreWebView2ContextMenuItem ** contextItem);
+
+        /// Moves the iterator to the next ContextMenuItem
+        HRESULT MoveNext([out, retval] BOOL * hasNext);
+
+    }
+
+    [uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+    interface ICoreWebView2_4 : ICoreWebView2_3
+    {
+        /// Add an event handler for the ContextMenuRequested event.
+        /// ContextMenuRequested event is raied when the user right clicks in the webview
+        ///
+        /// The host can use their own UI to create their own context menu using
+        /// the data provided in the API or can add to / remove from the default
+        /// context menu. provide a response with credentials for the authentication
+        /// or cancel the request. If the host doesn't handle the event, Webview will
+        /// display the original context menu.
+        ///
+        HRESULT add_ContextMenuRequested(
```

Comment:
Will talk to him and follow up with what he says

Reply:
After talks with Brad and Cagri, it seems like ICoreWebView2 is more appropriate since the context menus will be specifically handled by the app and not any framework.